### PR TITLE
Hook Redux up to the v2 keypad

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
The idea of this PR/ticket is to connect the v2 keypad to Redux so we can leverage existing state management.

One of the things I realized while doing this PR is that GestureManager is really only equipped to handle mobile (all event callbacks are for touch). Either we're going to have to replace GestureManager or we're going to have to get it working with a mouse/keyboard.

So far this PR:
- Wraps v2 keypad in a provider (`StatefulKeypad`)
  - Adds a story for this
- Lets `StatefulKeypad` accept a key press callback
- Connects v2 keypad button to the gesture manager for key presses

Issue: LC-803

## Test plan:

### CODE REVIEW

File: packages/math-input/src/components/keypad/stateful-keypad.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {action} from "@storybook/addon-actions";
+import * as React from "react";
+
+import StatefulKeypad from "./stateful-keypad";
+
+export default {
+    title: "Stateful v2 Keypad",
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/stateful-keypad.stories.tsx:7]: This should be `Perseus/Editor/Stateful v2 Keypad` so that it goes into the right place in the Storybook "menu", I think.

File: packages/math-input/src/components/keypad/button.tsx [hunk 2]
```diff
@@ -44,60 +49,76 @@ const styles = StyleSheet.create({
...
+                    </View>
+                </View>
             </View>
         );
     }
 }
+
+function mapStateToProps(state: State) {
+    return {
+        gestureManager: state.gestures.gestureManager,
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/button.tsx:120]: The more I see this the more I think that if anything, this Gesture Manager should go in React context.

File: packages/math-input/src/components/keypad/button.tsx [hunk 3]
```diff
@@ -20,22 +22,25 @@ const styles = StyleSheet.create({
...
+            boxShadow: "none",
+        },
+
+        ":focus": {
+            border: `2px solid ${Color.blue}`,
+            padding: 0,
+            boxShadow: "none",
+        },
+
+        ":active": {
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/button.tsx:38]: So we are replacing all the Redux tracking of these states with simple CSS pseudo-classes?  Awesome!!

File: packages/math-input/src/components/gesture-manager.ts [hunk 4]
```diff
@@ -25,7 +26,7 @@ class GestureManager {
         this.swipeEnabled = swipeEnabled;
 
         // Events aren't tracked until event tracking is enabled.
-        this.trackEvents = false;
+        this.trackEvents = true;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/gesture-manager.ts:30]: Was this change intentional?

File: packages/math-input/src/components/keypad/button.tsx [hunk 5]
```diff
@@ -44,60 +49,76 @@ const styles = StyleSheet.create({
...
+export type Props = OwnProps & ReduxProps;
+
+class Button extends React.Component<Props> {
     render(): React.ReactNode {
-        const {onPress, children, style, tintColor} = this.props;
+        const {children, style, tintColor, gestureManager, keyConfig} =
+            this.props;
+        const hovered = false;
+        const focused = false;
+        const pressed = false;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/button.tsx:81]: These look unused now as they trigger use of styles that don't even exist anymore (such as `styles.outerBoxHover`)

File: packages/math-input/src/components/keypad/button.tsx [hunk 6]
```diff
@@ -44,60 +49,76 @@ const styles = StyleSheet.create({
...
+};
+
+type ReduxProps = {
+    gestureManager: GestureManager;
 };
 
-export default class Button extends React.Component<Props> {
+export type Props = OwnProps & ReduxProps;
+
+class Button extends React.Component<Props> {
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/button.tsx:75]: FWIW, given this only has a `render` function, it could itself before a simple pure functional component.

File: packages/math-input/src/components/keypad/keypad-page-items.tsx [hunk 7]
```diff
@@ -43,6 +43,7 @@ export const KeypadButton = ({
     style,
 }: KeypadButtonProps): React.ReactElement => (
     <Button
+        keyConfig={keyConfig}
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad/keypad-page-items.tsx:46]: **suggestion**: It looks like all we use in the Button is the button ID.   I'd be inclined to just pass that in instead of the entire `keyConfig`.

**Comment by handeyeco** [comment]
Decided to not use Redux in v2 keypad
